File: backend/src/main/java/com/sprouthub/sprouthub/service/MarketplaceService.java
package com.sprouthub.sprouthub.service;

import com.sprouthub.sprouthub.model.Product;

import java.util.List;

public interface MarketplaceService {
    Product createProduct(Product product);
    Product getProductById(String id);
    List<Product> getAllProducts();
    Product updateProduct(String id, Product product);
    void deleteProduct(String id);
}

File: frontend/src/services/marketplaceService.js
import api from '../utils/api';

const getAllProducts = () => {
  return api.get('/marketplace');
};

const getProductById = (id) => {
  return api.get(`/marketplace/${id}`);
};

const createProduct = (data) => {
  return api.post('/marketplace', data);
};

const updateProduct = (id, data) => {
  return api.put(`/marketplace/${id}`, data);
};

const deleteProduct = (id) => {
  return api.delete(`/marketplace/${id}`);
};

const marketplaceService = {
  getAllProducts,
  getProductById,
  createProduct,
  updateProduct,
  deleteProduct,
};

export default marketplaceService;